warehouse/oidc/models/_core.py

```diff
@@ -102,6 +102,39 @@ def check_existing_jti(
     return True
 
 
+def verify_url_from_reference(reference_url: str, url: str) -> bool:
```

Comment:
I think it might make sense to make these keyword-only parameters, to force the caller to specify which one is the reference and which one is the URL to check (both are strings, so the order could be mistakenly inverted and there would be no warning/error)

```suggestion
def verify_url_from_reference(*, reference_url: str, url: str) -> bool:
```